src/sentry/sentry_metrics/configuration.py

```diff
@@ -43,8 +48,40 @@ def get_ingest_config(use_case_key: UseCaseKey) -> MetricsIngestConfiguration:
                 output_topic=settings.KAFKA_SNUBA_GENERIC_METRICS,
                 use_case_id=UseCaseKey.PERFORMANCE,
                 internal_metrics_tag="perf",
-                writes_limiter_cluster_options=settings.SENTRY_METRICS_INDEXER_WRITES_LIMITER_OPTIONS_PERFORMANCE,
+                writes_limiter_namespace=RateLimiterNamespace.PERFORMANCE,
             )
         )
 
     return _METRICS_INGEST_CONFIG_BY_USE_CASE[use_case_key]
+
+
+@dataclass(frozen=True)
+class RateLimiterConfiguration:
+    namespace: RateLimiterNamespace
+    writes_limiter_cluster_options: Mapping[str, Any]
+
+
+_WRITES_RATELIMITER_BY_NAMESPACE: MutableMapping[
+    RateLimiterNamespace, RateLimiterConfiguration
+] = dict()
```

Comment:
What's the purpose of having this additional mapping here instead of just including it in the main configuration object? Since there is 1:1 relationship between configuration object, rate limiter namespace and rate limiter configuration, I feel that having this separate mapping adds unnecessary indirection and makes things more complex because of all the additional registering overhead.

Reply:
okay so my understanding from https://github.com/getsentry/sentry/pull/37745#discussion_r944052716 was that although there is a 1:1 relationship right now, there might not be in the future? But maybe i misunderstood that comment.

Reply:
I don't follow, what else would it be if not 1:1? Surely we don't want to share namespaces nor ever assume configuration is shared.

Reply:
no I don't think we'd share namespaces, I was more thinking if there would ever be a case for multiple ratelimiting namespaces for a single backend. anyhoo i dk if that is realistic, I cant really think of a case rn but that's what I _thought_ filippo's comment was suggesting, but maybe what he was saying is that there should be a 1:1 relationship but it's not obvious. 

Regardless, if this was all combined within the one metrics ingest configuration object, are you suggesting the namespace is the lookup key ? instead of the `use_case_id` / the potential `(use_case_id, db_backend)`?

Reply:
No, I think you can still keep whatever lookup key strings you like, especially if you want the existing ones to be backwards compatible. Just don't have 2 mappings, merge them all into one.

Reply:
Sorry, let me clarify my comment on the other PR. 
What I meant was that it is not guaranteed that the DB_NAME would correspond to the rate limiter name space. Example: we may introduce another spanner backed indexer for the same use case that is based on a different spanner configuration. We would need a different rate limiter name space and we could not rely on the DB_NAME. 

More in general the issue here was caused by using the DB name to represent something that is not the DB but that, just as a coincidence, in the configuration you wrote, happened to be identifiable safely with the DB name.
Seeing that we are giving the rate limiter namespace a name that comes from an identifier of the DB should ring a bell and signal that something is wrong. 

Still one configuration will contain a rate limiter config and a DB name. never more than one.

So I would agree with Lyn's idea. You can remove this indirection and fir the `RateLimiterConfiguration` inside the `MetricsIngestConfiguration` object statically. Removing this indirection will make things easier to read (in this case as we are talking about a configuration, which is statically initialized). The structure is not complex and having it all visible in the same place overweights value of avoiding the duplication.